scib/metrics/morans_i.py

```diff
@@ -0,0 +1,149 @@
+import scanpy as sc
+import numpy as np
+import pandas as pd
+
+from scib.preprocessing import hvg_batch
+
+
+def morans_i(
+    adata_pre,
+    adata_post,
+    batch_key,
+    n_hvg=1000,
+    embed="X_pca",
+    rescale=True,
+    compare_pre=False,
+    hvgs=None,
+):
+    """
+    Compute Moran's I on HVGs (defined across batches) on integrated data
+        as a measure of bio conservation. The metric can be computed either as
+        A.) mean difference between Morans's I on embedding of integrated data and
+        max Moran's I of individual non-integrated batches or
+        B.) mean Morans's I on embedding of integrated data.
+    :param adata_pre: Non-integrated data.
+        Embedding is computed on scaled (m=0, s=1) X (expression)
+        per batch (recomputing HVGs, scaling, pca, neighbours).
+        For Moran's I unscaled X is used.
+    :param adata_post: Integrate data. Should have precomputed embedding and expression in X.
+    :param batch_key: Batch obs col in adata_pre.
+    :param n_hvg: How many top HVGs to use for Moran's I computation.
+    :param embed: Embedding to use for computation of neighbours in adata_post
+    :param rescale: Whether to rescale result to [0,1] so that 1 is better bio conservation
+    :param compare_pre: Whether to compute metric under scenario A instead of B.
+    :param hvgs: Custom list of genes to use for Moran's I computation (instead of hvgs
+        computed across batches).
+    :return: The resulting metric.
+    """
+    # Prepare adatas
+    # Get integrated connectivities for Moran's I
+    adata_post = adata_post.copy()
+    sc.pp.neighbors(adata_post, use_rep=embed)
+    # Prepare pre data
+    adata_pre = adata_pre.copy()
+    adata_pre.obs[batch_key] = adata_pre.obs[batch_key].astype("category")
```

Comment:
Please add a call to `check_sanity()` or the `check_adata` or `check_batch` util functions from here: https://github.com/theislab/scib/blob/f35b0c990f435e1c4bfab4a6fb38451712d010f6/scib/utils.py#L25